mkosi/config.py

```diff
@@ -1809,44 +1810,153 @@ def parse(self, argv: Optional[Sequence[str]] = None) -> tuple[MkosiArgs, tuple[
 
 
 class GenericVersion:
+    _EQUAL = 0
+    _RIGHT_SMALLER = 11
+    _LEFT_SMALLER = 12
```

Comment:
Why 11 and 12? Shouldn't we follow C here and do 0, -1 and 1?

Reply:
Here because 
> In the first mode — when OP is not specified — it will compare the two version strings and print either "VERSION1 < VERSION2", or "VERSION1 == VERSION2", or "VERSION1 > VERSION2" as appropriate.
>
> The exit status is 0 if the versions are equal, 11 if the version of the right is smaller, and 12 if the version of the left is smaller. (This matches the convention used by rpmdev-vercmp.)

and I wanted to make the `compare_versions` method behave like `systemd-analyze compare-versions` with only two arguments.

Reply:
Can we add a comment mentioning that?

Reply:
Added a comment.

Reply:
So after thinking a bit more, I think we shouldn't bother with 11 and 12 unless we need compat with rpmdev-vercmp somewhere, which I don't think we need. So I'd just use -1, 1 and 0 here instead.